modules/java/generator/android/java/org/opencv/android/OpenCVLoader.java.in

```diff
@@ -97,7 +95,7 @@
     /**
      * Current OpenCV Library version
      */
-    public static final String OPENCV_VERSION = BuildConfig.VERSION_NAME;
+    public static final String OPENCV_VERSION = "@OPENCV_VERSION_MAJOR@.@OPENCV_VERSION_MINOR@.@OPENCV_VERSION_PATCH@";
```

Comment:
With you suggestion https://github.com/opencv/opencv/pull/16222#discussion_r361693341 we do now https://github.com/hannesa2/owncloud-android/blob/70b27be1ade7f30b0e1e8e699ec6a61b09949aaf/owncloudApp/build.gradle#L263-L266 a calculation https://github.com/opencv/opencv/blob/24ca53183da18df02abf01058ea0927dff77f821/modules/java/android_sdk/android_gradle_lib/build.gradle#L4 
which we don't do here. So it differs !

That's why I tried to go into the direction of a single point of truth, instead to set this value on various places.

In an ideal world you would take the git tag name like https://github.com/hannesa2/owncloud-android/blob/70b27be1ade7f30b0e1e8e699ec6a61b09949aaf/owncloudApp/build.gradle#L263-L266

Reply:
Not sure that I understand you correctly.
`OPENCV_VERSION` ("4.2.0") != `package version code` (40020000)

Yes, Javadoc generation can be migrated on Gradle (for Android only? because we still have normal "desktop" Java with "Ant"): https://docs.gradle.org/current/dsl/org.gradle.api.tasks.javadoc.Javadoc.html
(unfortunatelly we don't have extra time for this task right now)